src/main/java/com/noxag/newnox/textanalyzer/algorithms/RepetitivWordingAnalyzer.java

```diff
@@ -1,18 +1,106 @@
+    }
+
+    private List<Finding> getRepetitionsInWordBlock(List<PDFPage> pages) {
+        List<TextPositionSequence> wordInPDF = PDFTextExtractionUtil.extractWords(pages);
+        List<Finding> findings = new ArrayList<>();
+
+        Map<String, Integer> repetitivWordMap = new HashMap<>();
+        List<TextPositionSequence> actuallyWordBlock = new ArrayList<>();
+
+        wordInPDF.stream().filter(word -> !PDFTextAnalyzerUtil.isPunctuationMark(word) && !isInteger(word.toString()))
+                .forEach(nextWord -> {
+                    if (actuallyWordBlock.size() < AMOUNT_OF_WORDS_TO_COMPARE) {
+                        repetitivWordMap.put(nextWord.toString(), calculateValueOfKey(nextWord, repetitivWordMap));
+                        actuallyWordBlock.add(nextWord);
+                    } else {
+                        removeKeyOutOfMap(repetitivWordMap, actuallyWordBlock.get(0).toString());
+                        repetitivWordMap.put(nextWord.toString(), calculateValueOfKey(nextWord, repetitivWordMap));
+                        actuallyWordBlock.remove(0);
+                        actuallyWordBlock.add(nextWord);
+                    }
+                    findings.addAll(addMultiplesToFindings(repetitivWordMap, actuallyWordBlock));
+                });
+        return findings;
+    }
+
+    private Integer calculateValueOfKey(TextPositionSequence nextWord, Map<String, Integer> repetitivWordMap) {
+        return (repetitivWordMap.containsKey(nextWord.toString()) ? repetitivWordMap.get(nextWord.toString()) : 0) + 1;
+    }
+
+    private void removeKeyOutOfMap(Map<String, Integer> map, String key) {
+        map.put(key, map.get(key) - 1);
+        if (map.get(key) == 0) {
+            map.remove(key);
+        }
+    }
+
+    private List<Finding> addMultiplesToFindings(Map<String, Integer> repetitivWordMap,
+            List<TextPositionSequence> actuallyWordBlock) {
+        List<Finding> findings = new ArrayList<>();
+        repetitivWordMap.entrySet().stream().filter(entry -> entry.getValue() > ALLOWED_REPETITIONS_BY_DEFAULT)
+                .forEach(repetitiveEntry -> {
+                    findings.addAll(getAllPositionSequencesThroughString(actuallyWordBlock, repetitiveEntry));
+                });
+        return findings;
+    }
+
+    private List<Finding> getAllPositionSequencesThroughString(List<TextPositionSequence> actuallyWordBlock,
+            Entry<String, Integer> repetitiveEntry) {
+        List<Finding> foundPositionSequences = new ArrayList<>();
+        actuallyWordBlock.stream().filter(wordsOfBlock -> wordsOfBlock.toString().equals(repetitiveEntry.getKey()))
+                .forEach(entryPositionSequence -> {
+                    foundPositionSequences
+                            .add(new TextFinding(entryPositionSequence, TextFindingType.REPETITIV_WORDING));
+                });
+        return foundPositionSequences;
+    }
+
+    private boolean isInteger(String string) {

```

Comment:
https://stackoverflow.com/questions/5439529/determine-if-a-string-is-an-integer-in-java